platform-service/src/unstract/platform_service/main.py

```diff
@@ -500,11 +512,62 @@ def custom_tool_instance() -> Any:
     return "Method Not Allowed", 405
 
 
-@app.errorhandler(CustomException)
-def handle_custom_exception(error: Any) -> tuple[Response, Any]:
-    response = jsonify({"error": error.message})
-    response.status_code = error.code  # You can customize the HTTP status code
-    return jsonify(response), error.code
+def log_exceptions(e: HTTPException) -> None:
+    """Helper method to log exceptions.
+
+    Args:
+        e (HTTPException): Exception to log
+    """
+    code = 500
+    if hasattr(e, "code"):
+        code = e.code or code
+
+    if code >= 500:
+        message = "{method} {url} {status}\n\n{error}\n\n````{tb}````".format(
+            method=request.method,
+            url=request.url,
+            status=code,
+            error=str(e),
+            tb=traceback.format_exc(),
+        )
+    else:
+        message = "{method} {url} {status} {error}".format(
+            method=request.method,
+            url=request.url,
+            status=code,
+            error=str(e),
+        )
+    app.logger.error(message)
+
+
+@app.errorhandler(HTTPException)
+def handle_http_exception(e: HTTPException) -> Response:
+    """Return JSON instead of HTML for HTTP errors."""
+    log_exceptions(e)
+    if isinstance(e, APIError):
+        return jsonify(e.to_dict()), e.code
```

Comment:
@chandrasekharan-zipstack Looks like a new sonar issue is introduced here.

![image](https://github.com/user-attachments/assets/170169d9-0211-499f-aac6-240efc3cc3bb)

Reply:
Apparently flask is takes care of these types with `Response` itself - will update the type hint